BASE/src/Microsoft.ApplicationInsights/Extensibility/TelemetryConfiguration.cs

```diff
@@ -334,6 +335,11 @@ public string ConnectionString
         /// </summary>
         public TelemetrySink DefaultTelemetrySink => this.telemetrySinks.DefaultSink;
 
+        /// <summary>
+        /// Gets an envelope for Azure.Core.TokenCredential which provides an AAD Authenticated token.
+        /// </summary>
+        internal ReflectionCredentialEnvelope CredentialEnvelope { get; private set; }
```

Comment:
I think you can still define an interface containing the GetToken() method, and this field be of that type. If we ever want to change the implementation from ReflectionCredentialEnvelope to something else, it'd be easy that way.

Reply:
done